packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -64,9 +65,72 @@ export enum ActionPartChange {
 	SAFE_CHANGE = 1,
 }
 
+export class DatastoreActionExecutionContext
+	extends ShowStyleUserContext
+	implements IDataStoreActionExecutionContext, IEventContext
+{
+	protected readonly _context: JobContext
+
+	constructor(
+		contextInfo: UserContextInfo,
+		context: JobContext,
+		showStyle: ReadonlyDeep<ShowStyleCompound>,
+		showStyleBlueprintConfig: ProcessedShowStyleConfig,
+		watchedPackages: WatchedPackagesHelper
+	) {
+		super(
+			contextInfo,
+			context.studio,
+			context.getStudioBlueprintConfig(),
+			showStyle,
+			showStyleBlueprintConfig,
+			watchedPackages
+		)
+		this._context = context
+	}
+
+	async setTimelineDatastoreValue(key: string, value: unknown, mode: 'temporary' | 'indefinite'): Promise<void> {
+		const collection = this._context.directCollections.TimelineDatastores
+		const entry = await collection.findOne({ studioId: this._context.studioId, key })
```

Comment:
If I've understood this correctly: There will only be one entry per `studio+key`?

If so, a few notes:
1. The `_id`  should not be a random string, but rather be a unique hash consisting of the studioId+key.
2. This ^ gives: the update could be done using an `upsert` instead, since that would reduce the number of db-calls.

```typescript
const id = protectedString(`${studioId}_${key}`)
collection.upsert(id, {$set: {
	_id: id,
	studioId: this._context.studioId,

	key,
	value,

	modified: Date.now(),
	mode,
}})
```